[AI gateway] Variety of tweaks / reformatting and reorganizing (#14674)
cloudflare/cloudflare-docs: content/ai-gateway/configuration/rate-limiting.md

---
pcx_content_type: configuration
title: Rate limiting
weight: 3
---

# Rate limiting

Rate limiting controls the traffic that reaches your application, which prevents expensive bills and suspicious activity.

## Parameters

> You can define rate limits as the number of requests that get sent in a specific time frame. For example, I can limit my application to 100 requests per 60s.

Reviewer: ```suggestion
You can define rate limits as the number of requests that get sent in a specific time frame. For example, you can limit your application to 100 requests per 60 seconds.
```

> You can also select if you would like a **fixed** or **sliding** rate limiting technique. With rate limiting, we allow a certain number of requests within a window of time. If it's fixed, the window would be based on time, so there would be no more than x requests in a 10 minute window (for example). If it's sliding, there would be no more than x requests in the last 10 minutes.

Reviewer: ```suggestion
You can also select if you would like a **fixed** or **sliding** rate limiting technique. With rate limiting, we allow a certain number of requests within a window of time. For example, if it is a fixed rate, the window is based on time, so there would be no more than `x` requests in a ten minute window. If it is a sliding rate, there would be no more than `x` requests in the last ten minutes.
```

> To illustrate this, let's say you had a limit of 10 requests per 10 minutes, starting at 12:00 so the fixed window is 12:00-12:10, 12:10-12:20, etc. If you sent 10 requests at 12:09 and 10 requests at 12:11, all 20 requests would be successful in a fixed window strategy, but would fail in a sliding window strategy since there were >10 requests in the last 10 minutes.

Reviewer: ```suggestion
To illustrate this, let us say you had a limit of ten requests per ten minutes, starting at 12:00. So the fixed window is 12:00-12:10, 12:10-12:20, and so on. If you sent ten requests at 12:09 and ten requests at 12:11, all 20 requests would be successful in a fixed window strategy. However, they would fail in a sliding window strategy since there were more than ten requests in the last ten minutes.
```

## Default configuration

{{<tabs labels="Dashboard | API">}}
{{<tab label="dashboard" no-code="true">}}

To set the default rate limiting configuration in the dashboard:

> 1. Log into the [Cloudflare dashboard](https://dash.cloudflare.com/).

Reviewer: ```suggestion
1. Log into the [Cloudflare dashboard](https://dash.cloudflare.com/) and select your account.
```

2. Go to **AI** > **AI Gateway**.
3. Go to **Settings**.

> 4. For **Rate-limiting**, switch the toggle to **On**.

Reviewer: ```suggestion
4. Enable **Rate-limiting**.
```
